src/compiler/checker.ts

```diff
@@ -16089,6 +16089,21 @@ namespace ts {
             return false;
         }
 
+        function getJavaScriptClassType(symbol: Symbol): Type | undefined {
+            if (symbol && isDeclarationOfFunctionOrClassExpression(symbol)) {
+                symbol = getSymbolOfNode((<VariableDeclaration>symbol.valueDeclaration).initializer);
+            }
+            if (isJavaScriptConstructor(symbol.valueDeclaration)) {
+                return getInferredClassType(symbol);
+            }
+            if (symbol.flags & SymbolFlags.Variable) {

```

Comment:
what's an example for this case? it looks like the `const B = function ...` case in the new test file, but I expected that to be imported from a new, third file instead of just being in the original file.